phone/src/apps/twitter/components/buttons/LikeButton.tsx

```diff
@@ -2,23 +2,35 @@ import React, { useState } from 'react';
 import { Button, CircularProgress } from '@mui/material';
 import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
 import FavoriteIcon from '@mui/icons-material/Favorite';
-import { useNuiRequest } from 'fivem-nui-react-lib';
 import { TwitterEvents } from '../../../../../../typings/twitter';
+import { fetchNui } from '../../../../utils/fetchNui';
+import { ServerPromiseResp } from '../../../../../../typings/common';
+import { useSnackbar } from '../../../../ui/hooks/useSnackbar';
+import { useTranslation } from 'react-i18next';
 
 const LOADING_TIME = 1250;
 
 function LikeButton({ tweetId, isLiked }) {
-  const Nui = useNuiRequest();
   const [liked, setLiked] = useState(isLiked);
   const [loading, setLoading] = useState(false);
+  const { t } = useTranslation();
+  const { addAlert } = useSnackbar();
 
   const handleClick = () => {
-    Nui.send(TwitterEvents.TOGGLE_LIKE, tweetId);
-    setLoading(true);
-    window.setTimeout(() => {
-      setLiked(!liked);
-      setLoading(false);
-    }, LOADING_TIME);
+    fetchNui<ServerPromiseResp<void>>(TwitterEvents.TOGGLE_LIKE, { tweetId }).then((resp) => {
+      if (resp.status !== 'ok') {
+        return addAlert({
+          message: t('APPS_TWITTER_LIKE_TWEET_FAILED'),
+          type: 'error',
+        });
+      }
+
+      setLoading(true);
+      window.setTimeout(() => {
+        setLiked(!liked);
+        setLoading(false);
```

Comment:
This seems a little bit odd as we are literally setting the loading state after we resolve the promise from the server. Also, this was utilized mostly for the previous fake loading system.